contrib/platform/test/com/sun/jna/platform/mac/CoreFoundationTest.java

```diff
@@ -0,0 +1,210 @@
+        cfMax.release();
+
+        IntByReference zero = new IntByReference(0);
+        IntByReference one = new IntByReference(1);
+        CFNumberRef cfZero = CF.CFNumberCreate(null, CFNumberType.kCFNumberIntType.ordinal(), zero);
+        CFNumberRef cfOne = CF.CFNumberCreate(null, CFNumberType.kCFNumberIntType.ordinal(), one);
+
+        assertEquals(0, cfZero.intValue());
+        assertEquals(1, cfOne.intValue());
+        cfZero.release();
+        cfOne.release();
+    }
+
+    @Test
+    public void testCFRetainCount() {
+        DoubleByReference pi = new DoubleByReference(Math.PI);
+        DoubleByReference e = new DoubleByReference(Math.E);
+        CFNumberRef cfE = CF.CFNumberCreate(null, CFNumberType.kCFNumberDoubleType.ordinal(), e);
+        CFNumberRef cfPi = CF.CFNumberCreate(null, CFNumberType.kCFNumberDoubleType.ordinal(), pi);
+        assertEquals(1, CF.CFGetRetainCount(cfE));
+        assertEquals(1, CF.CFGetRetainCount(cfPi));
+        cfE.retain();
+        cfPi.retain();
+        cfPi.retain();
+        assertEquals(2, CF.CFGetRetainCount(cfE));
+        assertEquals(3, CF.CFGetRetainCount(cfPi));
+
+        List<CFTypeRef> irrationalReferences = new ArrayList<>();
+        irrationalReferences.add(cfE);
+        irrationalReferences.add(cfPi);
+        for (CFTypeRef value : irrationalReferences) {
+            value.release();
+        }
+
+        assertEquals(1, CF.CFGetRetainCount(cfE));
+        assertEquals(2, CF.CFGetRetainCount(cfPi));
+        cfPi.release();
+        assertEquals(1, CF.CFGetRetainCount(cfPi));
+        cfE.release();
+        cfPi.release();
+    }
+
+    @Test
+    public void testCFArray() {
+        CFNumberRef[] refArray = new CFNumberRef[3];
+        int size = Native.getNativeSize(CFNumberRef.class);
+        Memory contiguousArray = new Memory(size * refArray.length);
+        for (int i = 0; i < refArray.length; i++) {
+            refArray[i] = CF.CFNumberCreate(null, CoreFoundation.CFNumberType.kCFNumberIntType.ordinal(),
+                    new IntByReference(i));
+            contiguousArray.setPointer(i * size, refArray[i].getPointer());
+        }
+        CFArrayRef cfPtrArray = CF.CFArrayCreate(null, contiguousArray, refArray.length, null);
+
+        assertEquals(refArray.length, CF.CFArrayGetCount(cfPtrArray));
+        for (int i = 0; i < refArray.length; i++) {
+            CFTypeRef result = CF.CFArrayGetValueAtIndex(cfPtrArray, i);
+            CFNumberRef numRef = new CFNumberRef(result.getPointer());
+            assertEquals(i, numRef.intValue());
+        }
+
+        for (int i = 0; i < refArray.length; i++) {
+            refArray[i].release();
+        }
+        cfPtrArray.release();
+    }
+
+    @Test
+    public void testCFData() {
+        String deadBug = "The only good bug is a dead bug.";
+        Memory bugBytes = new Memory(deadBug.length() + 1);
+        bugBytes.clear();
+        bugBytes.setString(0, deadBug);
```

Comment:
Beware - if the String contains multibyte characters, this will blow (a german umlaut will be encoded as 2 bytes if the native encoding is UTF-8). I would create a byte array on the java side.

Reply:
Good catch. I agree, when I initially changed the CString method I realized this but didn't think to check this other occurrence.

Reply:
Actually, this is fine. I only used a String as a convenient method of generating some bytes to fill the memory.  I switched this test to actually generate and read random bytes to make it more clear what's being tested.